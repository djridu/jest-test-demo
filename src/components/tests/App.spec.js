import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';
import Header from '../../containers/Header';
import MainSection from '../../containers/MainSection';

const setup = () => {
    const renderer = new ShallowRenderer();
    renderer.render(<App />);
    return renderer.getRenderOutput();
};

describe('components', () => {
    describe('<App />', () => {
        it('should render and match the snapshot', () => {
            const output = setup();
            expect(output).toMatchSnapshot();
        });
    });

    describe('<Header />', () => {
        it('should render', () => {
            const output = setup();
            const [header] = output.props.children;
            expect(header.type).toBe(Header);
        });
    });

    describe('<Mainsection />', () => {
        it('should render', () => {
            const output = setup();
            const [, mainSection] = output.props.children;
            expect(mainSection.type).toBe(MainSection);
        });
    });
});
