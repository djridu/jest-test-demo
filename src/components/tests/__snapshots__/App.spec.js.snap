// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <App /> should render and match the snapshot 1`] = `
<div>
  <Connect(Header) />
  <Connect(MainSection) />
</div>
`;
